apis/projectcontour/v1alpha1/contourdeployment.go

```diff
@@ -151,6 +156,16 @@ type EnvoySettings struct {
 	// More info: https://kubernetes.io/docs/concepts/configuration/manage-resources-containers/
 	// +optional
 	Resources corev1.ResourceRequirements `json:"resources,omitempty"`
+
+	// An update strategy to replace existing Envoy DaemonSet pods with new pods.
+	// when envoy be running as a `Deployment`,it's must be nil
+	// +optional
+	UpdateStrategy *appsv1.DaemonSetUpdateStrategy `json:"updateStrategy,omitempty"`
+
+	// Strategy describes the deployment strategy to use to replace existing Envoy pods with new ones.
+	// when envoy be running as a `DaemonSet`,it's must be nil
+	// +optional
+	Strategy *appsv1.DeploymentStrategy `json:"strategy,omitempty"`
```

Comment:
Perhaps this would be a good time to create subsections for DeploymentSettings vs. DaemonSetSettings, and then move these under their respective sections. We could also add `Replicas` to Deployment settings, and maybe have a one-release deprecation cycle for the existing Replicas setting. @sunjayBhatia WDYT? I'm not 100% sure it's worth it, but if we are going to do it, it's easier the sooner we do it.